Correct Service Bus tracing details. (#1017)
dotnet/docs-aspire: docs/messaging/azure-service-bus-component.md

| `ConnectionString`        | The connection string used to connect to the Service Bus namespace. |
| `Credential`              | The credential used to authenticate to the Service Bus namespace.   |
| `FullyQualifiedNamespace` | The fully qualified Service Bus namespace.                          |

> | `DisableHealthChecks`     | Disables health checks for the Service Bus client.                  |

Reviewer: This isn't correct. `AzureMessagingServiceBusSettings` has:

* public string? HealthCheckQueueName
* public string? HealthCheckTopicName

You need to set one of those properties in order to enable health checks.

Author: To confirm, both are not required, just one or the other?

Author: What happens if both are provided?

- `Azure.Core`
- `Azure.Identity`
- `Azure-Messaging-ServiceBus`

### Tracing

> To enable tracing, set the `DisableTracing` option to `false` in the configuration. When enabled, the .NET Aspire Azure Service Bus component will emit the following tracing activities using OpenTelemetry:

Reviewer: > To enable tracing, set the `DisableTracing` option to `false` in the configuration.

This isn't correct. From the XML docs:

```
    /// <remarks>
    /// ServiceBus ActivitySource support in Azure SDK is experimental, the shape of Activities may change in the future without notice.
    /// It can be enabled by setting "Azure.Experimental.EnableActivitySource" <see cref="AppContext"/> switch to true.
    /// Or by setting "AZURE_EXPERIMENTAL_ENABLE_ACTIVITY_SOURCE" environment variable to "true".
    /// </remarks>
```

- `Azure.Core`
- `Azure.Identity`
- `Azure-Messaging-ServiceBus`

### Tracing

> [!NOTE]
> Service Bus `ActivitySource` support in the Azure SDK for .NET is experimental, and the shape of activities may change in the future without notice.

> To enable tracing, you can do so in one of three ways:

Reviewer 2: ```suggestion
You can enable tracing in one of three ways:
```

> - Set the `DisableTracing` option to `false` in the configuration.

Reviewer: This won't do it. The default is `false`. So setting it to `false` won't do anything.

> - Call `System.AppContext.SetSwitch("Azure.Experimental.EnableActivitySource", true)`.

Reviewer: ````suggestion
- Setting the `Azure.Experimental.EnableActivitySource` [runtime configuration setting](https://learn.microsoft.com/dotnet/core/runtime-config/) to `true`. Which can be done with either:
    - Call `AppContext.SetSwitch("Azure.Experimental.EnableActivitySource", true);`.
    - Add the RuntimeHostConfigurationOption setting to your project file:
```xml
<ItemGroup>
    <RuntimeHostConfigurationOption Include="Azure.Experimental.EnableActivitySource" Value="true" />
</ItemGroup>
```
````

Reviewer: See https://devblogs.microsoft.com/azure-sdk/introducing-experimental-opentelemetry-support-in-the-azure-sdk-for-net/#get-started

Reviewer 3: FWIW https://github.com/davidfowl/Build2024AspireDemo/blob/main/AppHost/ServiceBusTracingExtensions.cs

- `Azure.Core`
- `Azure.Identity`
- `Azure-Messaging-ServiceBus`

### Tracing

> [!NOTE]
> Service Bus `ActivitySource` support in the Azure SDK for .NET is experimental, and the shape of activities may change in the future without notice.

You can enable tracing in one of three ways:

> - Set the `DisableTracing` option to `false` in the configuration.

Author: ```suggestion
```
